### PR TITLE
Add config to change the delay before sending a notification email

### PR DESCRIPTION
**Pull request by MatMaul**
Signed-off-by: Mathieu Velten <mathieu.velten@beta.gouv.fr>

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog).
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

File: docs/usage/configuration/config_documentation.md [hunk 1]
```diff
@@ -663,6 +663,8 @@ This setting has the following sub-options:
    has missed. Disabled by default.
 * `notif_for_new_users`: Set to false to disable automatic subscription to email
    notifications for new users. Enabled by default.
+* `delay_before_mail`: Time we always wait before ever emailing about a notification
+   (to give the user a chance to respond to other push or notice the window). Defaults to 10 minutes.
```
Comment (clokep) [inline docs/usage/configuration/config_documentation.md:667]:
```suggestion
* `delay_before_mail`: The time to wait before emailing about a notification.
  This gives the user a chance to view the message via push or an open client.
  Defaults to 10 minutes.

   *New in Synapse 1.98.0._
```

Should this be `notif_delay_before_mail`? Some of other settings aren't related to notifications.

File: synapse/config/emailconfig.py [hunk 2]
```diff
@@ -294,6 +294,9 @@ def read_config(self, config: JsonDict, **kwargs: Any) -> None:
             self.email_riot_base_url = email_config.get(
                 "client_base_url", email_config.get("riot_base_url", None)
             )
+            self.delay_before_mail_ms = Config.parse_duration(
+                email_config.get("delay_before_mail", "10m")
+            )
```
Comment (clokep) [inline synapse/config/emailconfig.py:299]: I think keeping the comments from the constant is useful:

```suggestion
            # The amount of time we always wait before ever emailing about a notification
            # (to give the user a chance to respond to other push or notice the window)
            self.delay_before_mail_ms = Config.parse_duration(
                email_config.get("delay_before_mail", "10m")
            )
```

**Review by clokep — COMMENTED** [review]
Seems reasonable, but a couple small comments.

**Comment by MatMaul** [comment]
We would also need some configuration nobs to tweak throttle max and throttle multiplier, but I was thinking about doing that in another PR. I can add that to this one if you prefer.

**Review by erikjohnston — APPROVED** [review]
Thanks @MatMaul!

**Pull request merged**